src/include/parser/create_statement.h

```diff
@@ -329,35 +296,27 @@ struct ColumnDefinition {
  */
 class IndexAttr {
  public:
-  /**
-   * Create an index attribute on a column name.
-   */
+  /** Create an index attribute on a column name. */
   explicit IndexAttr(std::string name) : name_(std::move(name)), expr_(nullptr) {}
 
-  /**
-   * Create an index attribute on an expression.
-   */
-  explicit IndexAttr(std::shared_ptr<AbstractExpression> expr) : name_(""), expr_(std::move(expr)) {}
+  /** Create an index attribute on an expression. */
+  explicit IndexAttr(common::ManagedPointer<AbstractExpression> expr) : name_(""), expr_(expr) {}
 
-  /**
-   * @return the name of the column that we're indexed on
-   */
+  /** @return the name of the column that we're indexed on */
   std::string GetName() const {
     TERRIER_ASSERT(expr_ == nullptr, "Expressions don't come with names.");
     return name_;
   }
 
-  /**
-   * @return the expression that we're indexed on
-   */
-  std::shared_ptr<AbstractExpression> GetExpression() const {
+  /** @return the expression that we're indexed on */
+  common::ManagedPointer<AbstractExpression> GetExpression() const {
     TERRIER_ASSERT(expr_ != nullptr, "Names don't come with expressions.");
-    return expr_;
+    return common::ManagedPointer(expr_);
```

Comment:
why the wrapper, isn't expr_ already a managed pointer?

Reply:
Good catch.